patches/server/1047-Add-FluidState-API.patch

```diff
@@ -0,0 +1,222 @@
+new file mode 100644
+index 0000000000000000000000000000000000000000..cfabb814ebd281aab299c6c655266ff357e08806
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/block/fluid/package-info.java
+@@ -0,0 +1,5 @@
++@DefaultQualifier(NonNull.class)
++package io.papermc.paper.block.fluid;
++
++import org.checkerframework.checker.nullness.qual.NonNull;
++import org.checkerframework.framework.qual.DefaultQualifier;
+diff --git a/src/main/java/io/papermc/paper/block/fluid/type/PaperFallingFluid.java b/src/main/java/io/papermc/paper/block/fluid/type/PaperFallingFluid.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..bf4b5e1b1c1cd13ed65c3b3fe1af13b8101e8d97
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/block/fluid/type/PaperFallingFluid.java
+@@ -0,0 +1,18 @@
++
++package io.papermc.paper.block.fluid.type;
++
++import io.papermc.paper.block.fluid.PaperFluidData;
++import net.minecraft.world.level.material.FlowingFluid;
++import net.minecraft.world.level.material.FluidState;
++
++public class PaperFallingFluid extends PaperFluidData implements FallingFluid {
++
++    public PaperFallingFluid(final FluidState state) {
++        super(state);
++    }
++
++    @Override
++    public boolean isFalling() {
++        return this.get(FlowingFluid.FALLING);
++    }
++}
+diff --git a/src/main/java/io/papermc/paper/block/fluid/type/PaperFlowing.java b/src/main/java/io/papermc/paper/block/fluid/type/PaperFlowing.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..017fe1e6892f1cd06e6a9beee53a91efa183be5e
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/block/fluid/type/PaperFlowing.java
+@@ -0,0 +1,27 @@
++package io.papermc.paper.block.fluid.type;
++
++import net.minecraft.world.level.material.FlowingFluid;
++import net.minecraft.world.level.material.FluidState;
++import org.bukkit.util.Vector;
++
++public class PaperFlowing extends PaperFallingFluid implements Flowing {
++
++    public PaperFlowing(final FluidState state) {
++        super(state);
++    }
++
++    @Override
++    public int getLevel() {
++        return this.get(FlowingFluid.LEVEL);
++    }
++
++    @Override
++    public int getMinimumLevel() {
++        return FlowingFluid.LEVEL.min;
++    }
++
++    @Override
++    public int getMaximumLevel() {
++        return FlowingFluid.LEVEL.max;
++    }
++}
+diff --git a/src/main/java/org/bukkit/craftbukkit/CraftRegionAccessor.java b/src/main/java/org/bukkit/craftbukkit/CraftRegionAccessor.java
+index 1a17875426468b287c8ea3f559ea516d0218f7a0..0bd63e98501fcfbfddadda9bd5b7e9340dbf3950 100644
+--- a/src/main/java/org/bukkit/craftbukkit/CraftRegionAccessor.java
++++ b/src/main/java/org/bukkit/craftbukkit/CraftRegionAccessor.java
+@@ -202,6 +202,7 @@ import org.bukkit.inventory.ItemStack;
+ import org.bukkit.potion.PotionData;
+ import org.bukkit.potion.PotionType;
+ import org.bukkit.util.Vector;
++import org.jetbrains.annotations.NotNull;
```

Comment:
Inline the import (or just don't use it at all, not needed)

Reply:
Oh yeah oops forgot about fully qualified imports